frank-flow/src/frontend/src/app/editor/explorer/add-dialog/add-dialog.component.ts

```diff
@@ -28,8 +28,62 @@ export class AddDialogComponent {
     this.currentDirectory = this.codeService.currentDirectory;
   }
 
+  findDuplicateFileName(fileName: string, directory: any): boolean {
+    if (directory != null && fileName != null) {
+      const foundFile = directory._files.find(
+        (file: string) => file === fileName
+      );
+      if (foundFile) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  findCurrentDirectory(directory: any, path: string): any {
+    if (path === '') {
+      return directory;
+    } else {
+      let nextDirRegex = path.match(/.*(?=\/)/g);
+      let nextDir;
+      let nextPath = path.replace(/.*\//g, '');
+
+      if (!nextDirRegex) {
+        nextDir = path;
+        nextPath = '';
+      } else {
+        nextDir = nextDirRegex[0];
+      }
+
+      if (nextDir) {
+        return this.findCurrentDirectory(directory[nextDir], nextPath);
+      }
+    }
+  }
+
+  checkForDuplicateFiles(): boolean {
+    const fileName = this.fileName;
+    const rootDir = this.fileService.configurationFiles.value;
+
+    if (this.currentDirectory.path !== undefined) {
+      const currentDirectory = this.findCurrentDirectory(
+        rootDir[0].content,
+        this.currentDirectory.path
+      );
+
+      if (this.findDuplicateFileName(fileName, currentDirectory)) {
+        this.toastr.error(
+          `The file ${fileName} already exists in this directory.`,
+          'Error creating'
+        );
+        return true;
+      }
+    }
+    return false;
```

Comment:
Same story. Try to return the statement. You can move the toastr to the function where this check is used.

Reply:
Changed the function to return the statement. I chose to let the toastr stay in the 'checkForDuplicateFiles()' function in case we need to check for duplicates in the future. Also it would make the 'add()' function larger which is already pretty large.